k8s/k8s.go

```diff
@@ -272,8 +274,9 @@ type RBACOps interface {
 
 // PodOps is an interface to perform k8s pod operations
 type PodOps interface {
-	// CreatePod creates the given pod
-	CreatePod(pod *v1.Pod) (*v1.Pod, error)
+	// CreatePod creates the given pod. If update is true, the API updates the given pod
+	// instead of creating a new pod
+	CreatePod(pod *v1.Pod, update bool) (*v1.Pod, error)
```

Comment:
Every other API here has a separate UpdateObject API, we should keep it consistent

Reply:
done